src/API/Site/Controllers/MerchantCenter/PolicyComplianceCheckController.php

```diff
@@ -0,0 +1,168 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\CountryCodeTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\EmptySchemaPropertiesTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Google\GoogleHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\WC;
+use WC_Install;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class PolicyComplianceCheckController
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter
+ */
+class PolicyComplianceCheckController extends BaseController {
+
+	use CountryCodeTrait;
+	use EmptySchemaPropertiesTrait;
+
+	/**
+	 * The WC proxy object.
+	 *
+	 * @var WC
+	 */
+	protected $wc;
+
+	/**
+	 * @var GoogleHelper
+	 */
+	protected $google_helper;
+
+	/**
+	 * BaseController constructor.
+	 *
+	 * @param RESTServer   $server
+	 * @param WC           $wc
+	 * @param GoogleHelper $google_helper
+	 */
+	public function __construct( RESTServer $server, WC $wc, GoogleHelper $google_helper ) {
+		parent::__construct( $server );
+		$this->wc            = $wc;
+		$this->google_helper = $google_helper;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		$this->register_route(
+			'mc/policy_check/allowed_countries',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_allowed_countries(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+			]
+		);
+
+		$this->register_route(
+			'mc/policy_check/payment_gateways',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->has_payment_gateways(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+			]
+		);
+
+		$this->register_route(
+			'mc/policy_check/store_ssl',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_is_store_ssl(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+			]
+		);
+
+		$this->register_route(
+			'mc/policy_check/refund_return_policy',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->has_refund_return_policy_page(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get the allowed countries for the controller.
+	 *
+	 * @return array
+	 */
+	protected function get_allowed_countries(): array {
```

Comment:
I was reading the PRD, but I can't find anything related to allowed countries, how is this going to be used? Shouldn't this be allowed to countries by GMC?

Reply:
We are planning to check if the store is live and accessible to all users. There is one check for this policy which is to check for access geo restrictions. Then we would like to check if the store set up the allowed countries. Please correct me if I'm wrong.